python-sdk/src/astro/databases/google/bigquery.py

```diff
@@ -254,30 +258,10 @@ def create_table_using_native_schema_autodetection(
         :param table: The table to be created.
         :param file: File used to infer the new table columns.
         """
-        load_job_config = {
-            "sourceUris": [file.path],
-            "destinationTable": {
-                "projectId": self.get_project_id(table),
-                "datasetId": table.metadata.schema,
-                "tableId": table.name,
-            },
-            "sourceFormat": NATIVE_PATHS_SUPPORTED_FILE_TYPES[file.type.name],
-            "autodetect": True,
-        }
-
-        job_config = {
-            "jobType": "LOAD",
-            "load": load_job_config,
-            "labels": {"target_table": table.name},
-        }
-
-        self.hook.insert_job(
-            configuration=job_config,
+        supported_config = NATIVE_AUTODETECT_SCHEMA_CONFIG.get(
+            file.location.location_type
         )
-
-        # We have to clear the table afterwards as bigquery automatically loads the data when creating the table.
-        statement = f"TRUNCATE TABLE `{self.get_table_qualified_name(table)}`"
-        self.run_sql(statement)
```

Comment:
Thanks @utkarsharma2. The truncate statement was pretty ugly and probably also error prone (potentially vulnerable to race conditions). And of course speed and cost plays an import role when loading data into the data warehouse.

Reply:
I am still wondering how I could come up with that idea 🤦